### PR TITLE
feat: s/u filter

### PR DESCRIPTION
**Pull request by Leomotors**
## Why did you create this PR

- Add S/U Filter

## Demo

[https://dev.cugetreg.com](https://dev.cugetreg.com)

### CODE REVIEW

File: apps/web/src/modules/CourseSearch/components/FilterSection/constants.tsx [hunk 1]
```diff
@@ -26,6 +26,17 @@ export const createGenEdCheckboxes: CreateCheckbox<GenEdChipKey>[] = [
   },
 ]
 
+export const createGradingCheckboxes: CreateCheckbox<GradingChipKey>[] = [
+  {
+    label: 'S/U Grade',
```
Comment (bombnp) [inline apps/web/src/modules/CourseSearch/components/FilterSection/constants.tsx:31]: Use `S/U` to be consistent with course detail page
```suggestion
    label: 'S/U',
```

File: apps/web/src/modules/CourseSearch/hooks/useSearchCourseQueryParams/utils/extractSearchVarsFromQuery/index.ts [hunk 2]
```diff
@@ -9,14 +9,16 @@ export function extractSearchVarsFromQuery(
   query: QueryParams,
   courseGroup: CourseGroup
 ): SearchCourseQueryVariables {
-  const { keyword, genEdTypes, dayOfWeeks, startTime, endTime } = query
+  const { keyword, genEdTypes, gradingTypes, dayOfWeeks, startTime, endTime } = query
 
   const genEdTypeArray = genEdTypes ? genEdTypes.split(',') : undefined
+  const gradingTypeArray = gradingTypes ? gradingTypes.split(',') : undefined
```
Comment (bombnp) [inline apps/web/src/modules/CourseSearch/hooks/useSearchCourseQueryParams/utils/extractSearchVarsFromQuery/index.ts:15]: im pretty sure we can use this instead
```suggestion
  const gradingTypeArray = gradingTypes?.split(',')
```

File: apps/web/src/common/components/Chips/config.tsx [hunk 3]
```diff
@@ -33,6 +33,13 @@ export const genEdChipConfig: GenEdChipConfigProps = {
   NO: { label: 'ไม่ใช่ Gened', color: 'deepGrayOutlined', variant: 'outlined' },
 }
 
+export type GradingChipKey = GradingType
+export type GradingChipConfigProps = Record<GradingChipKey, ChipConfigProps>
+export const gradingChipConfig: GradingChipConfigProps = {
+  S_U: { label: 'S/U Grade', color: 'blueOutlined', variant: 'outlined' },
```
Comment (bombnp) [inline apps/web/src/common/components/Chips/config.tsx:39]: Use `S/U` to be consistent with course detail page
```suggestion
  S_U: { label: 'S/U', color: 'blueOutlined', variant: 'outlined' },
```

**Review by bombnp — COMMENTED** [review]
other than that, LGTM

File: apps/web/src/common/components/Chips/config.tsx (hunk 3 above)
Comment (Leomotors) [reply apps/web/src/common/components/Chips/config.tsx:39]: I decided to use S/U Grade in Course Detail Page instead so it align with "Letter Grade"

Comment (bombnp) [reply apps/web/src/common/components/Chips/config.tsx:39]: image
It's still S/U here na

Comment (Leomotors) [reply apps/web/src/common/components/Chips/config.tsx:39]: Yes, will update in this PR

File: apps/web/src/modules/CourseSearch/hooks/useSearchCourseQueryParams/utils/extractSearchVarsFromQuery/index.ts (hunk 2 above)
Comment (Leomotors) [reply apps/web/src/modules/CourseSearch/hooks/useSearchCourseQueryParams/utils/extractSearchVarsFromQuery/index.ts]: Done, also updated other similar variables.

**Pull request merged**